### PR TITLE
♻️ Refactor `SimpleLogReg`

### PR DESCRIPTION
**Pull request by falexwolf**
Produce a first sensible run for LogReg on a small Tahoe100M subset through fixing basic bugs and adding loss logging.

## Summary

Training F1:

* pytorch: 0.081
* scikit-learn: 0.131

Discrepancy likely is due to supoptimal hyperparameters. Next version.

## Pytorch

https://lamin.ai/laminlabs/arrayloader-benchmarks/transform/djTEQo0BLWCI0001

image

## scikit-learn

https://lamin.ai/laminlabs/arrayloader-benchmarks/transform/6tvhWWfp40Yh0001

image

## Materials

Internal [Slack thread](https://laminlabs.slack.com/archives/C08FJFAV0H1/p1753185595719759).

### CODE REVIEW

File: modlyn/models/linear.py [hunk 1]
```diff
@@ -1,51 +0,0 @@
...
-    def forward(self, inputs):
-        return self.linear(inputs)
-
-    def training_step(self, batch, batch_idx):
-        x, targets = batch
-        logits = self.forward(x)
-        preds = torch.argmax(logits, dim=1)
-        loss = F.cross_entropy(logits, targets)
-        self.log("train_loss", loss)
-        metrics = self.train_metrics(targets, preds)
```
Comment (falexwolf) [inline modlyn/models/linear.py:33]: The order here was inverted, so nothing could have worked @mikelkou @felix0097
